### ISSUE TITLE
Deprecating a field on an InputObjectGraphType does not reflect on the Introspection query response.

### ISSUE THREAD

**Issue by BatScream** [graphql-dotnet/graphql-dotnet#1112]
## Description

This issue is related to the below issues:

https://github.com/graphql/graphql-spec/issues/197 (open)

The Introspection query response does not include the `isDeprecated` and `deprecationReason` fields for `InputFields` marked as deprecated.

### Steps to reproduce

This is a simple mutation in test:

                FieldAsync<SampleResponse>(
                    "SampleMutation",
                    arguments: new QueryArguments(
                        new QueryArgument<SampleInput>
                        {
                            Name = "Input"
                        }
                    ),
                    resolve: async context =>
                    {
                        return null;
                    }
                );

The below is the query argument type with a list of integers that is marked as deprecated.

```
    public class SampleInput : InputObjectGraphType
    {
        public SampleInput()
        {
            Name = "SampleInput";
           Field<ListGraphType<IntGraphType>>("someIntList", deprecationReason: "Deprecated").
        }
   }
```

The introspection query fired by the GraphQL client does not include the `isDeprecated` and `deprecationReason` for  `InputFields` (Looks like this is not even supported).

```
query IntrospectionQuery {
  __schema {
    queryType {
      name
    }
    mutationType {
      name
    }
    subscriptionType {
      name
    }
    types {
      ...FullType
    }
    directives {
      name
      description
      locations
      args {
        ...InputValue
      }
    }
  }
}

fragment FullType on __Type {
  kind
  name
  description
  fields(includeDeprecated: true) {
    name
    description
    args {
      ...InputValue
    }
    type {
      ...TypeRef
    }
    isDeprecated
    deprecationReason
  }
  inputFields {
    ...InputValue
  }
  interfaces {
    ...TypeRef
  }
  enumValues(includeDeprecated: true) {
    name
    description
    isDeprecated
    deprecationReason
  }
  possibleTypes {
    ...TypeRef
  }
}

fragment InputValue on __InputValue {
  name
  description
  type {
    ...TypeRef
  }
  defaultValue
}

fragment TypeRef on __Type {
  kind
  name
  ofType {
    kind
    name
    ofType {
      kind
      name
      ofType {
        kind
        name
        ofType {
          kind
          name
          ofType {
            kind
            name
            ofType {
              kind
              name
              ofType {
                kind
                name
              }
            }
          }
        }
      }
    }
  }
}
```

### Expected result

The introspection query response should get the below subset of the response representing `sampleInput` and thus enabling to query on the `isDeprecated` and `deprecationReason` fields on  `InputFields`.

```
        {
          "kind": "INPUT_OBJECT",
          "name": "SampleInput",
          "description": null,
          "fields": null,
          "inputFields": [
            {
              "name": "SampleInput",
              "description": null,
              "isDeprecated": true,
              "deprecationReason": "Deprecated",
              "type": {
                "kind": "LIST",
                "name": null,
                "ofType": { "kind": "SCALAR", "name": "Int", "ofType": null }
              },
              "defaultValue": "null"
            }
          ],
          "interfaces": null,
          "enumValues": null,
          "possibleTypes": null
        }
```

### Actual result

But the actual subset of the response obtained from the server on the introspection query is as below (without the `isDeprecated` and `deprecationReason` fields.)

```
        {
          "kind": "INPUT_OBJECT",
          "name": "SampleInput",
          "description": null,
          "fields": null,
          "inputFields": [
            {
              "name": "SampleInput",
              "description": null,
              "type": {
                "kind": "LIST",
                "name": null,
                "ofType": { "kind": "SCALAR", "name": "Int", "ofType": null }
              },
              "defaultValue": "null"
            }
          ],
          "interfaces": null,
          "enumValues": null,
          "possibleTypes": null
        }
```

### Environment

.Net Core 2.1
GraphQL version 2.4.0
graphql-playground-react@1.7.10

**Comment by joemcbride** [comment]
At first I was thinking perhaps this was a bug in this project's introspection, however it appears that the GraphQL specification does not currently provide the ability to deprecate Input Objects / Values.

See these two issues:
https://github.com/graphql/graphql-spec/issues/197
https://github.com/graphql/graphql-spec/pull/525

So in that regard, the bug in this project is that it allows you provide a deprecation reason for Inputs.

**Comment by BatScream** [comment]
Thank you. Yes that was indeed confusing. It would be better if we remove the deprecation reason for inputs, to be in sync with the specification.

**Comment by sungam3r** [comment]
> It would be better if we remove the deprecation reason for inputs, to be in sync with the specification

It will be difficult with the current organization of classes. In addition, the draft specification has a proposal graphql/graphql-spec#525 to add deprecations for inputs. I hope that we will live to see when this comes true.

**Comment by sungam3r** [comment]
Can be made as part of #1451

**Comment by sungam3r** [comment]
@BatScream this will be published in v7.4
